Bottom: 843b269db50b197b8dde9a1c84a314e33bfeb77f
Top:    dc2232cbe4b5f59458b29964d9cf41ff87ed4e3b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-15 00:24:17 +0100

report.c: Don't leave actions in the undo stack (and not increment the serial)

Some actions in ReportNetLengthByName() would have been lumped in with
whatever actions were next added to the undo stack.



---

diff --git a/src/report.c b/src/report.c
index 4d5e27b..07d7d94 100644
--- a/src/report.c
+++ b/src/report.c
@@ -761,6 +761,7 @@ ReportNetLengthByName (char *tofind, int x, int y)
   LibraryMenuType *net;
   ConnectionType conn;
   int net_found = 0;
+  int retval = 0;
 #if defined(USE_RE)
   int use_re = 0;
 #endif
@@ -778,11 +779,8 @@ ReportNetLengthByName (char *tofind, int x, int y)
   if (!tofind)
     return 1;
 
-  SaveUndoSerialNumber ();
-  ResetFoundPinsViasAndPads (true);
-  RestoreUndoSerialNumber ();
-  ResetFoundLinesAndPolygons (true);
-  RestoreUndoSerialNumber ();
+  ResetConnections (true);
+  IncrementUndoSerialNumber ();
 
 #if defined(USE_RE)
       use_re = 1;
@@ -805,14 +803,16 @@ ReportNetLengthByName (char *tofind, int x, int y)
 	      regerror (result, &elt_pattern, errorstring, 128);
 	      Message (_("regexp error: %s\n"), errorstring);
 	      regfree (&elt_pattern);
-	      return (1);
+              retval = 1;
+              goto out;
 	    }
 #endif
 #if defined(HAVE_RE_COMP)
 	  if ((elt_pattern = re_comp (tofind)) != NULL)
 	    {
 	      Message (_("re_comp error: %s\n"), elt_pattern);
-	      return (false);
+              retval = 1;
+              goto out:
 	    }
 #endif
 	}
@@ -862,8 +862,10 @@ ReportNetLengthByName (char *tofind, int x, int y)
   if (!net_found)
     {
       gui->log ("No net named %s\n", tofind);
-      return 1;
+      retval = 1;
+      goto out;
     }
+
 #ifdef HAVE_REGCOMP
   if (use_re)
     regfree (&elt_pattern);
@@ -872,16 +874,16 @@ ReportNetLengthByName (char *tofind, int x, int y)
   length = XYtoNetLength (x, y, &found);
   netname = net->Name + 2;
 
-  if (!found && net_found)
-  {
-      gui->log ("Net found, but no lines or arcs were flagged.\n");
-      return 1;
-  }
-  else if (!found)
-  {
-      gui->log ("Net not found.\n");
-      return 1;
-  }
+  if (!found)
+    {
+      if (net_found)
+        gui->log ("Net found, but no lines or arcs were flagged.\n");
+      else
+        gui->log ("Net not found.\n");
+
+      retval = 1;
+      goto out;
+    }
 
   {
     char buf[50];
@@ -891,7 +893,12 @@ ReportNetLengthByName (char *tofind, int x, int y)
     else
       gui->log ("Net length: %s\n", buf);
   }
-  return 0;
+  retval = 0;
+
+out:
+  Undo (true);
+
+  return retval;
 }
 
 /* ---------------------------------------------------------------------------
